bin/cli.js

```diff
@@ -69,6 +86,22 @@ function errorAndExit(errorMessage) {
 async function generateSchema(pathToSpec) {
   const output = cli.flags.output ? OUTPUT_FILE : OUTPUT_STDOUT; // FILE or STDOUT
 
+  // Parse incoming headers from CLI flags
+  let httpHeaders = {};
+  // prefer --headersObject if specified
+  if (cli.flags.headersObject) {
+    httpHeaders = JSON.parse(cli.flags.headersObject); // note: this will generate a recognizable error for the user to act on
+  }
+  // otherwise, parse --header
+  else if (Array.isArray(cli.flags.header)) {
+    cli.flags.header.forEach((header) => {
+      const firstColon = header.indexOf(":");
```

Comment:
This was using a RegEx instead (`/(\w+)(\:)\s((\w+)[\-|\:]?.+)/gi;`), but in testing I found it to be brittle and a little awkward. Instead I opted for more of a `curl`-like approach where the first `:` designates the key–value relationship. It’s then up to the user to format their header correctly.